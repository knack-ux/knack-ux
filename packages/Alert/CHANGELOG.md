# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.6.6](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.5...@knack-ux/alert@0.6.6) (2020-02-12)

**Note:** Version bump only for package @knack-ux/alert





## [0.6.5](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.4...@knack-ux/alert@0.6.5) (2020-01-23)

**Note:** Version bump only for package @knack-ux/alert





## [0.6.4](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.3...@knack-ux/alert@0.6.4) (2020-01-20)

**Note:** Version bump only for package @knack-ux/alert





## [0.6.3](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.2...@knack-ux/alert@0.6.3) (2019-12-21)

**Note:** Version bump only for package @knack-ux/alert





## [0.6.2](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.1...@knack-ux/alert@0.6.2) (2019-12-21)

**Note:** Version bump only for package @knack-ux/alert





## [0.6.1](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.6.0...@knack-ux/alert@0.6.1) (2019-12-21)

**Note:** Version bump only for package @knack-ux/alert





# [0.6.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.5.0...@knack-ux/alert@0.6.0) (2019-12-14)


### Features

* **icon, avatar:** create components ([0901f92](https://github.com/knack-ux/knack-ux/commit/0901f92))





# [0.5.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.4.0...@knack-ux/alert@0.5.0) (2019-12-09)


### Features

* **packages:** remove most of them clean ([06aa5a9](https://github.com/knack-ux/knack-ux/commit/06aa5a9))





# [0.4.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/alert@0.3.0...@knack-ux/alert@0.4.0) (2019-11-28)


### Features

* **package.json:** add repository fields ([36c89ad](https://github.com/knack-ux/knack-ux/commit/36c89ad))





# 0.3.0 (2019-11-26)


### Features

* **many:** rewrite some components ([800dab3](https://github.com/chrispcode/knack/commit/800dab3))





# 0.2.0 (2019-09-22)


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))
