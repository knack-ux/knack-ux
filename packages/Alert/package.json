{
  "name": "@knack-ux/alert",
  "version": "0.3.0",
  "description": "Alert component for the Knack UI Project",
  "main": "lib/index.js",
  "repository" : {
    "type" : "git",
    "url": "ssh://git@github.com/knack-ux/knack-ux.git",
    "directory": "@knack-ux/alert"
  },
  "author": "Kristiyan Serafimov (chrispcode)",
  "license": "MIT",
  "scripts": {
    "start": "webpack --watch --config=../../webpack/webpack.dev.js",
    "build": "webpack --config=../../webpack/webpack.prod.js",
    "build:doc": "webpack --config=../../webpack/webpack.dev.js",
    "clean": "rm -rf lib"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "peerDependencies": {
    "framer-motion": "^1.6.8",
    "react": "^16.8.4",
    "styled-components": "^4.1.3",
    "styled-system": "5.1.2"
  }
}
