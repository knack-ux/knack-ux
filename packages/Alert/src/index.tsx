import React, { HTMLAttributes } from 'react';
import {
  system, LayoutProps, SpaceProps
} from 'styled-system';

import {
  Container,
  Description,
  Title,
  Wrap
} from './styled';

type ReactDiv = HTMLAttributes<HTMLDivElement>;

export interface Props extends ReactDiv, LayoutProps, SpaceProps {
  title?: string
  description: string
  intent?: string
}

export function Alert({
  title,
  description,
  intent = 'info',
  ...props
}: Props) {
  function renderIcon() {
    let iconName = intent;
    if (intent === 'danger') iconName = 'error';
    else if (intent === 'success') iconName = 'check';

    return (
      <i className="material-icons-round">
        {iconName}
      </i>
    );
  }

  return (
    <Wrap intent={intent} {...props}>
      {renderIcon()}
      <Container>
        {title && <Title>{title}</Title>}
        <Description>
          {description}
        </Description>
      </Container>
    </Wrap>
  );
}


export default Alert;
