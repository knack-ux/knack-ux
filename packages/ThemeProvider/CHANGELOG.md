# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.7.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.6.0...@knack-ux/theme@0.7.0) (2019-12-21)


### Features

* **badge:** create badge component ([36bd405](https://github.com/knack-ux/knack-ux/commit/36bd405))





# [0.6.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.5.0...@knack-ux/theme@0.6.0) (2019-12-14)


### Features

* **icon, avatar:** create components ([0901f92](https://github.com/knack-ux/knack-ux/commit/0901f92))





# [0.5.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.4.0...@knack-ux/theme@0.5.0) (2019-12-09)


### Features

* **packages:** remove most of them clean ([06aa5a9](https://github.com/knack-ux/knack-ux/commit/06aa5a9))





# [0.4.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.3.0...@knack-ux/theme@0.4.0) (2019-11-28)


### Features

* **package.json:** add repository fields ([36c89ad](https://github.com/knack-ux/knack-ux/commit/36c89ad))





# 0.3.0 (2019-11-26)


### Bug Fixes

* **input:** typings ([2b922ad](https://github.com/chrispcode/knack/commit/2b922ad))


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **many:** rewrite some components ([800dab3](https://github.com/chrispcode/knack/commit/800dab3))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))





# 0.2.0 (2019-09-22)


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))
