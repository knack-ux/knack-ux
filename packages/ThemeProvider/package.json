{
  "name": "@knack-ux/theme",
  "version": "0.2.0",
  "description": "Theme provider for the Knack UI Project",
  "main": "lib/index.js",
  "repository": "https://github.com/chrispcode/knack",
  "author": "Kristiyan Serafimov (chrispcode)",
  "license": "MIT",
  "scripts": {
    "start": "webpack --watch --config=../../webpack/webpack.dev.js",
    "build": "webpack --config=../../webpack/webpack.prod.js",
    "build:doc": "webpack --config=../../webpack/webpack.dev.js",
    "clean": "rm -rf lib"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "peerDependencies": {
    "react": "^16.8.4",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "lodash.get": "^4.4.2"
  }
}
