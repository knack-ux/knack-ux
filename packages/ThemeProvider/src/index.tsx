import React from 'react';
import { ThemeProvider as Base, DefaultTheme } from 'styled-components';
import _get from 'lodash.get';


// and extend them!


export const defaultTheme = {
  a11y: {
    default: '#0052CC',
    info: '#0052CC',
    success: '#00855f',
    warning: '#B15f00',
    danger: '#D63B00'
  },
  colors: {
    default: '#0052CC',
    info: '#0052CC',
    success: '#3DA68C',
    warning: '#FF8B00',
    danger: '#FF5630'
  },
  focus: '#CED6E0'
};


declare module 'styled-components' {
  export interface DefaultTheme {
    a11y: {
      default: string
      success: string
      warning: string
      danger: string
    },
    colors: {
      default: string
      success: string
      warning: string
      danger: string
    },
    focus: string
  }
}

export function ThemeProvider({
  children
}: any) {
  return (
    <Base theme={defaultTheme}>
      {children}
    </Base>
  );
}

export default ThemeProvider;
