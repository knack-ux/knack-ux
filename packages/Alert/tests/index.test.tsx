import React from 'react';
import { render } from '@testing-library/react';

import Alert from '../src';
import ThemeProvider from '../../ThemeProvider';

describe('<Alert />', () => {
  test('snapshots', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Alert
          title="Error 500"
          description="Something went wrong!"
          intent="danger"
        />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
