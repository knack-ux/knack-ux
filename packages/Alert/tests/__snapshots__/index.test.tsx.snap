// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> snapshots 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm SnzzN"
  >
    <i
      class="src__IconBase-m4bsf3-0 fhDrnd material-icons-round"
      font-size="24"
    >
      error
    </i>
    <div
      class="sc-AxhCb ccMvAG"
    >
      <div
        class="sc-AxirZ fRWfze"
        font-size="2,3"
      >
        Error 500
      </div>
      <div
        class="sc-AxiKw cAQzyg"
        font-size="1,2"
      >
        Something went wrong!
      </div>
    </div>
  </div>
</DocumentFragment>
`;
